src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -303,7 +304,11 @@ public MapBean readMap(Path mapFolder) throws MapLoadException {
   @Cacheable(value = CacheNames.MAP_PREVIEW, unless = "#result == null")
   public Image loadPreview(String mapName, PreviewSize previewSize) {
     if (mapGeneratorService.isGeneratedMap(mapName)) {
-      return mapGeneratorService.getGeneratedMapPreviewImage();
+      try {
+        return new Image(new FileInputStream(forgedAlliancePreferences.getCustomMapsDirectory().resolve(mapName).resolve(mapName + "_preview.png").toFile()));
```

Comment:
Can u not do new Image(....getURL())

Reply:
There is no url for this image as this is getting the image generated on the local filesystem

Reply:
Hard to read. My proposal:

```
Path previewPath = forgedAlliancePreferences.getCustomMapsDirectory().resolve(mapName).resolve(mapName + "_preview.png");
return new Image(Files.newInputStream(previewPath));
```

I'm also not totally sure when the exception would be thrown? If the file does not exist yet? That shouldn't be catched via try-catch but with `Files.exists(previewPath)` instead (-> never use Exceptions for control flow).
If it's used for anything else, please add a comment.